### ISSUE TITLE
FRAG-12 blows up explosives on someone's person, even with a shielded hardsuit

### ISSUE THREAD

**Issue by carlarctg** [tgstation/tgstation#46822]
## Reproduction:
1. Equip shielded hardsuit
2. Put on grenade belt
3. get shot by FRAG-12
4. Explode

This is an issue even without the shielded suit bug, as it means any random joe with an improvised shotgun can onehitkill any nukie with an easy-to-manufacture custom shell.

**Comment by Tlaltecuhtli** [comment]
any explosion will trigger every explosive in your inventory not a bug

**Comment by necromanceranne** [comment]
I think it's because the projectile is supposed to be completely negated by a shielded hardsuit, yet the explosions still trigger. Which is probably actually a bug. You shouldn't be blown up by something that was prevented by wearing the shielded hardsuit.

**Comment by Tlaltecuhtli** [comment]
should shield suits make you immune to any explosion?

**Comment by necromanceranne** [comment]
Should an explosion occur if the bullet that would explode gets deleted by the shield?

**Comment by Tlaltecuhtli** [comment]
does it delete rockets or gyro gun stuff?

**Comment by necromanceranne** [comment]
I wouldn't think so, but then similar things like sand golems absorbing bullets will delete rockets as well as likely frag 12 if it is at all consistent.

So I guess what is the intended behaviour? Because frankly I can definitely see justification for this to be intended behavior, but it isn't consistent with a few other similar effects.

**Comment by yoyobatty** [comment]
Probably could add protect_contents or set contents_explosion to return false for your character whenever your armor stops damage. Maybe this could be turned into a component?

**Comment by carlarctg** [comment]
stop saying 'not a bug' and ignoring the issue, issues are issues even if they aren't bugs reeee

frag-12, instantly killing any half-prepared nukie seems like really bad balance to me, even if it does need corn oil to make you can just do that yourself in the garden while grabbing the rest of the materials for shells

**Comment by 81Denton** [comment]
What @yoyobatty said sounds really good. Since shielded hardsuits cost 30TC and only have three charges, I'd also be fine with them completely blocking explosions as long as the shield is up.

Anyway, this is something for after the antag freeze.

**Comment by kriskog** [comment]
@81Denton You know the shield recharges right?

**Comment by 81Denton** [comment]
Yeah, but you can also quickly kill the shield with buckshot etc. and then have 20 seconds before it starts recharging.

**Comment by carlarctg** [comment]
the shield is very good, but buckshot erases it instantly, and half the crew runs combat shotguns with buckshot in warops. At that point it feels like there's no reason to take it when the elite hardsuit only costs 8 tc and is more effective and blocking buckshot.

having a 30tc counter to the incredibly frustrating suicidebombs that have permeated every warops round i've been in would be, in my opinion, a great idea

**Comment by Sadboysuss** [comment]
Shielded hardsuit doesn't exist anymore, and afaik grenade belts don't explode this way anymore

**Comment by necromanceranne** [comment]
I think I'll test it and see whats happening. While shielded hardsuits don't exist, we do have the shield from the suit still in the game as the shield module.

Since this issue though, that got completely refactored. So who knows.

**Comment by necromanceranne** [comment]
From testing; it doesn't pop your contained belt bombs anymore. But it definitely pierces shields still.

https://github.com/tgstation/tgstation/blob/6f2fde52180cd189c2eb423c681cafaf1eabe4ed/code/modules/projectiles/projectile/bullets/shotgun.dm#L60-L69

Most likely culprit is that it's not checking ``blocked``.
